src/xmpp/iq.c

```diff
@@ -2761,7 +2761,7 @@ _mam_rsm_id_handler(xmpp_stanza_t* const stanza, void* const userdata)
                 // Convert to iso8601
                 start_str[strlen(start_str) - 3] = '\0';
             }
-            auto_char char* end_str = NULL;
```

Comment:
Ouch yes! Missed that in the last review. Because it is actually data->end_datestr.

Reply:
nope, it isn't. It's free'd inside `log_database_get_previous_chat()`